vcd/resource_vcd_nsxt_edgegateway_dns_test.go

```diff
@@ -0,0 +1,586 @@
+    name = "{{.DefaultForwarderName}}"
+    
+    upstream_servers = [
+      "{{.ServerIp1}}",
+      "{{.ServerIp2}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone1}}"
+
+    upstream_servers = [
+      "{{.ServerIp5}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName2}}",
+      "{{.DomainName3}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone2}}"
+
+    upstream_servers = [
+      "{{.ServerIp3}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName1}}",
+    ]
+  }
+}
+
+data "vcd_nsxt_edgegateway_dns" "data_{{.DnsConfig}}" {
+  edge_gateway_id = vcd_nsxt_edgegateway_dns.{{.DnsConfig}}.edge_gateway_id
+
+  depends_on = [vcd_nsxt_edgegateway_dns.{{.DnsConfig}}]
+}
+
+resource "vcd_nsxt_edgegateway_dns" "{{.VdcGroupDnsConfig}}" {
+  edge_gateway_id = data.vcd_nsxt_edgegateway.{{.EdgeGw}}.id
+  enabled         = true
+  
+  default_forwarder_zone {
+    name = "{{.DefaultForwarderName}}"
+    
+    upstream_servers = [
+      "{{.ServerIp1}}",
+      "{{.ServerIp2}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone1}}"
+
+    upstream_servers = [
+      "{{.ServerIp5}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName2}}",
+      "{{.DomainName3}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone2}}"
+
+    upstream_servers = [
+      "{{.ServerIp3}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName1}}",
+    ]
+  }
+}
+
+data "vcd_nsxt_edgegateway_dns" "data_{{.VdcGroupDnsConfig}}" {
+  edge_gateway_id = vcd_nsxt_edgegateway_dns.{{.VdcGroupDnsConfig}}.edge_gateway_id
+
+  depends_on = [vcd_nsxt_edgegateway_dns.{{.VdcGroupDnsConfig}}]
+}
+`
+
+func TestAccVcdNsxtEdgegatewayDnsIpSpaces(t *testing.T) {
+	preTestChecks(t)
+
+	if checkVersion(testConfig.Provider.ApiVersion, "<38.0") {
+		t.Skip("This test is only supported since version 38.1 of the API")
+	}
+
+	// String map to fill the template
+	var params = StringMap{
+		"TestName":             t.Name(),
+		"NsxtManager":          testConfig.Nsxt.Manager,
+		"NsxtTier0Router":      testConfig.Nsxt.Tier0router,
+		"Org":                  testConfig.VCD.Org,
+		"NsxtVdc":              testConfig.Nsxt.Vdc,
+		"EdgeGw":               "ip-spaces-egw",
+		"ExtNet":               t.Name(),
+		"DnsConfig":            t.Name(),
+		"DefaultForwarderName": t.Name() + "default",
+		"ServerIp1":            "1.1.1.1",
+		"ServerIp2":            "2.2.2.2",
+		"ServerIp3":            "3.3.3.3",
+		"IpAddress1":           "11.11.11.100",
+		"IpAddress2":           "11.11.11.101",
+	}
+	testParamsNotEmpty(t, params)
+
+	params["FuncName"] = t.Name() + "step1"
+	configText1 := templateFill(testAccVcdNsxtEdgegatewayDnsIpSpacesStep1, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s\n", configText1)
+
+	params["FuncName"] = t.Name() + "step2"
+	configText2 := templateFill(testAccVcdNsxtEdgegatewayDnsIpSpacesStep2, params)
```

Comment:
Test in step 1 and step 2 are similar. They are needed in this framework, but it would be a waste in binary tests. Please add a skip statement for binary test in step 1 and remove it for step 2.

Reply:
Fixed.